### DISCUSSION TITLE
Stealth mode for HTTP proxy (HTTPPROXY_STEALTH)

### DISCUSSION THREAD

**Discussion by pokemaster974** [qdm12/gluetun#1921] [Q&A]
Hi, I've already installed gluetun but now I'm actually trying to create a HTTP proxy.
For now, I'm not certain of this option in [https://github.com/qdm12/gluetun-wiki/blob/main/setup/options/http-proxy.md](url)
> "Stealth mode means HTTP proxy headers are not added to your requests"

Is it better to let it OFF (like as default) or is it better to switch it to ON ?
What's the pros and the cons of the 2 differents options ?

Regards.

**Comment by oalieno** [comment]
https://github.com/qdm12/gluetun/blob/6ac7ca4f0fb66a96a62d9c5e2df85eb9fb85e0d7/internal/httpproxy/http.go#L28
https://github.com/qdm12/gluetun/blob/6ac7ca4f0fb66a96a62d9c5e2df85eb9fb85e0d7/internal/httpproxy/http.go#L62-L72

According to the code, stealth mode means DO NOT add your ip to the `X-Forwarded-For` header. `X-Forwarded-For` is a HTTP header that contains multiple ips like `1.2.3.4, 2.3.4.5`. It is a list that tracks all the ip all the way from source to destination (there might be multiple proxy server along the way).

If you leave the default (OFF), the destination server will receive something like the following.

```
GET / HTTP/1.1
Host: 1.2.3.4
User-Agent: curl/7.88.1
Accept: */*
X-Forwarded-For: 172.16.0.1
Accept-Encoding: gzip
```

The ip appears to be private ip `172.16.0.1` because I sent the request from my host machine to the docker inside. But you see the idea, your ip will be pass in the HTTP header. I think most people use gluetun for privacy, likely *arr apps. My recommendation is to **turn it ON**, no harm in turning this on.
